### DISCUSSION TITLE
Creality touch screen support

### DISCUSSION THREAD

**Discussion by DevSecTim** [Jyers/Marlin#1084] [Q&A] [answered]
So installed Jyers firmware and loved it, so useful! Now installed Creality touch screen and it doesn't get past the empty progress bar on startup - I am a dev and might be able to get it working with the right pointers...?

Anyone?

**Comment by Reggi3** [comment]
Which LCD screen?  Do you know if there's a specific model of creality printer that uses this touchscreen as its stock LCD?  Have you checked the pinout for the LCD and compared it to the pinout for the LCD header on the 4.x.x mainboard?  If there is a specific creality printer then I would look at the configuration.h and configuration_adv.h for that model and see if you can pull the correct defines for your screen from them, then compare and combine to/with the JyersUI configs.  

The only real issue that I see, is if it's not a DWIN based LCD, that will 100% be why it's not working for you.  I guess it could also be down to whether the TS has been calibrated but you'd expect to see some on screen feedback for that.

**Comment by Reggi3** [comment]
Adding some more meat to the bones on this one, the touchscreen is DWIN but the graphics for it are somewhat different to the JyersUI graphics, it looks to me that without some fairly heavy coding intervention, it's not going to be touch usable :-(

From the official creality store, I followed the links for their instructions/firmware:
https://drive.google.com/drive/folders/1rB7KClgPikSu6m4FHUu6bCuZgwPjtjWw

If you check the 'installation instructions' folder and open the PDF, you'll se the graphical difference between the 2 LCDs.  If you look in the touchscreen folder, you'll notice that it's definitely DWIN based.  IMHO to work with JyersUI, marlin will need some work and the LCD firmware itself will need a reasonable amount of work (unless you're experienced in DWIN) to get it up to scratch.

**Reply by Reggi3** [reply]
It is absolutely not a DWIN screen ... (DWIN is a screen manufacturer!), could you explain that logic to me please?  I'm not entirely sure what point you're trying to make, so if you could elaborate a bit more that would be useful.  It seems that you might not have read my post properly or something was lost in translation perhaps?

I call it DWIN because that's how our LCD is referenced, throughout the source code:
https://github.com/Jyers/Marlin/tree/Ender_3_V2_JyersUI/Marlin/src/lcd/dwin
https://github.com/Jyers/Marlin/blob/Ender_3_V2_JyersUI/Marlin/Configuration.h#L2562

I could refer to dgus when talking about DWIN based lcd screens but that wouldn't make it easy to find in the context of e3v2 stock screens or the creality touchscreens in the source code...

**Comment by smooreace** [comment]
" I call it DWIN because that's how our LCD is referenced, throughout the
source code:"

Thats because the V2 screen is made by the company DWIN, and it functions
the DWIN designed it to function. The new touchscreen that Creality is
selling and packaging with some printers is not made by DWIN, and functions
the way its manufacturer designed it to function. Much like the fact that
you cant put a Ford starter on a GM car and expect it to work.

**Comment by smooreace** [comment]
BTW... I dont believe Creality "manufactures" anything. At best they pay
someone else to build their mainboards

**Comment by Reggi3** [comment]
Sigh, we can argue over what to call it as much as you like, This is the link to the OFFICIAL CREALITY firmware for this screen
https://drive.google.com/drive/folders/1BPQ_4gxbCWaJQCtVRjzBmU8NLwIDPLho

I found this link on this official creality page too:
https://www.creality3dofficial.com/products/upgraded-touch-panel-screen-display-kits-for-ender-3

Can you see what the filename is for the file in the top left?  0_DWIN_ASC.HZK  People seem to be confused as to why I call it DWIN, it's pretty simple, it's because it's named DWIN in any reference to source code and lcd firmware.

either way I stand by my statements, it's going to require source code for JyersUI firmware that isn't present, the same with the DWIN based touchscreen.  Both will require either some source code and a firmware update to run with JyersUI firmware.

**Reply by DevSecTim** [reply]
Hi folks thanks for the enthusiastic response! I will take a look at the screen and let you know the specific one. But it's official Creality regardless of who manufacture it (it comes in a Creality box etc).

**Comment by Jyers** [comment, marked as answer]
@tititopher68-dev @Reggi3 @smooreace 
I'm going to finalize this discussion on what it's called so there's no confusion floating around. 

The most accurate descriptor for the stock V2 lcd AND the newer touchscreens is DWIN. These are both DWIN screens. We call them this because they are made by DWIN, simple enough.

DGUS is not an identifier of any type. It is actually the name of the software used to program displays like the touchscreen. DGUS itself stands for **DWIN Graphic Utilized Software**. So while the use of the DGUS software is only needed for the new touchscreens, it would be a bit strange to then refer to them by DGUS.

Despite that it's in no way wrong to call them DGUS screens (I think marlin even throws that term around a bit) but that definitely does not mean that they are not DWIN screens. Both are 100% DWIN screens, made by DWIN.

**Reply by DevSecTim** [reply]
Interesting indeed, anything we can do to make the screen work in your awesome firmware?

**Reply by Jyers** [reply]
Basically the main issue is that the screen needs to have the UI programmed into it, rather than programmed on the mainboard... This makes it almost entirely incompatible with my current source. I'll definitely look into it in the future, but there's no 'quick fix' for making it work with the firmware.

**Reply by DevSecTim** [reply]
Ah I see fair enough, well I put the original screen back on anyway. Thanks for all the hard work, the firmware is really a lot better than stock, Z offset adjusting is a dream :-)

**Reply by Reggi3** [reply]
Another user asked about these touchscreens on discord, so I went on a little hunt to attempt to track down more information.  

https://github.com/InsanityAutomation/Marlin/tree/CrealityDwin_2.0
https://github.com/InsanityAutomation/Marlin/releases/tag/DW731

I think this might contain enough information to maybe get started?  there are DWIN_SET folders in the release.  The insanity automation marlin fork is pretty remarkable, they don't do different configs for each machine, it appears to be one big config that does 400+ combinations of hardware and looks a bit hairy at first glance.

**Reply by Jyers** [reply]
It's not a matter of getting started. I technically know everything that needs to be done/developed to get this working. I just don't have time right now to put in the work of creating the display firmware to utilize my UI. I've talked to insanity automated extensively and yeah he has a huge autobuild library for configs. His work in mainly geared towards the older touchscreens though, not the one we are talking about which is the CR-6 se touchscreen

**Reply by Reggi3** [reply]
I should've been more direct towards the OP @TurboTim I had hoped that it might've been enough for him to look at and get started on something.  You've said there's nothing in there for the cr-6 se touchscreen, so this information isn't useful, now I know...

**Reply by DevSecTim** [reply]
Thanks @reggi3 but sounds like @Jyers  has it under control, apart from needing more hours in the day
